ci_framework/plugins/modules/get_makefile_env.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Chandan Kumar <chkumar@redhat.com>
+# Apache License Version 2.0 (see LICENSE)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: get_makefile_env
+
+short_description: Gets Makefile default variables.
+
+description:
+- Retrieves a dictionary with all the environment variables
+  defined in the given Makefile.
+
+options:
+  makefile_path:
+      description:
+      - Full path of the the makefile.
+      required: True
+      type: str
+author:
+  - Chandan Kumar (@raukadah)
+  - Cedric Jeanneret (@cjeanner)
+  - Pablo Rodriguez (@pablintino)
+'''
+
+EXAMPLES = r'''
+# Gets /home/user/Makefile default variable values
+- name: Generate env var fact
+  get_makefile_env:
+    makefile_values: /home/user/Makefile
+'''
+
+RETURN = r'''
+makefile_values:
+    description: The Makefile variables and their default values
+    type: dict
+    returned: success
+    sample:
+        - NAMESPACE: openstack
+          NETWORK_ISOLATION: true
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+
+
+# Return install_yamls vars
+def __get_makefile_vars(makefile_path):
+    if not os.path.exists(makefile_path):
+        return None
+
+    with open(makefile_path) as f:
+        lines_split = [line.split('?=') for line in f.readlines() if '?=' in line and not line.startswith('#')]
```

Comment:
I don't think so since ?= is a conditional assignment, it assigns it's value only if the value isn't provided from outside i.e a param or an environment variable. := assignments will ignore our given values so I assume that those variables are the ones that are not meant to be overridden.